ra/registration-authority.go

```diff
@@ -680,7 +680,15 @@ func (ra *RegistrationAuthorityImpl) checkLimits(names []string, regID int64) er
 	if limits.CertificatesPerName.Enabled() {
 		err := ra.checkCertificatesPerNameLimit(names, limits.CertificatesPerName, regID)
 		if err != nil {
-			return err
+			if _, ok := err.(core.RateLimitedError); !ok {
+				return err
+			}
+			if exists, dbErr := ra.SA.FQDNSetExists(names); dbErr != nil {
```

Comment:
This took me a couple reads to comprehend - I think it should be split onto more lines.